src/App.tsx

```diff
@@ -1,27 +1,87 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum Modes {
+  'just5',
+  'onlyRed',
+  'all',
+  'none',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [mode, setMode] = useState<Modes>(Modes.none);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  useEffect(() => {
+    const loadAll = async () => {
+      const allGoods = await goodsAPI.getAll();
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      setGoods(allGoods);
+    };
 
-    <GoodsList goods={[]} />
-  </div>
-);
+    const loadFirst5Goods = async () => {
+      const first5Goods = await goodsAPI.get5First();
+
+      setGoods(first5Goods);
+    };
+
+    const loadRedGoods = async () => {
+      const getRedGoods = await goodsAPI.getRedGoods();
+
+      setGoods(getRedGoods);
+    };
+
+    switch (mode) {
+      case Modes.all:
+        loadAll();
+        break;
+
+      case Modes.just5:
+        loadFirst5Goods();
+        break;
+
+      case Modes.onlyRed:
+        loadRedGoods();
+        break;
+
+      default:
+        setGoods([]);
+    }
+  }, [mode]);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => setMode(Modes.all)}
+      >
+        Load all goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => setMode(Modes.just5)}
```

Comment:
I think it would be more concise and simpler if you just used the methods here. There is not any additional logic in switch cases, and doing these things with modes just creates some unnecessary in this case code. You can directly use these functions as you have them all defined anyway
```suggestion
        onClick={loadFirst5Goods}
```